detox/src/devices/common/drivers/android/tools/AppInstallHelper.js

```diff
@@ -18,6 +18,28 @@ class AppInstallHelper {
     const binaryPathOnTarget = await this._fileXfer.send(deviceId, binaryPath, binaryFilenameOnTarget);
     await this._adb.remoteInstall(deviceId, binaryPathOnTarget);
   }
+
+  async clearUserData(deviceId, bundleId) {
+    return await this._adb.clearUserData(deviceId, bundleId);
+  }
+
+  async getLocalBinaryHash(binary) {
+    const { getMd5 } = require('./CryptoUtils');
+    await getMd5(binary);
+  }
+
+  async isAlreadyInstalled(deviceId, filehash) {
+    return await this._fileXfer.checkFileExists(deviceId,`${filehash}.hash`);
+  }
+
+  async _removeExistingFileHashes(deviceId) {
+    return await this._fileXfer.deleteByExtension(deviceId, 'hash');
+  }
+
+  async recordHash(deviceId, filehash) {
+    await this._removeExistingFileHashes(deviceId);
+    await this._fileXfer.createEmptyFile(deviceId,`${filehash}.hash`);
+  }
```

Comment:
This doesn't seem associated with what the class is meant to do (namely, install APK's). You need to come up with a better design where this would all make sense. To be honest, I'm not sure this is needed at all...